Code/Query/EqualityQuery.h

```diff
@@ -23,15 +23,20 @@ template <typename MatchFuncArgType,
 class EqualityQuery
     : public Query<MatchFuncArgType, DataFuncArgType, needsConversion> {
  public:
-  EqualityQuery() : d_tol(0) { this->df_negate = false; };
+  EqualityQuery() {
```

Comment:
I know the base default constructor is called automatically, but I still like the explicit call in the initializer list.  This is stylistic though.

Reply:
I like the initializer list too, it's a pity that it can't be used with members of the base class.

Reply:
Ah, but you can!  Yay curly braces!

```class F {
public:
  int a;
  int b;
};

class Z : public F {
public:
   Z() : F{1,2} {}
};```

Reply:
I think you're supposed to provide all data members when you do that though.
And that would be tricky for us

Reply:
Agreed :)

Reply:
Yeah, I didn't noticed the protected variables, but I will note that df_negate should be initialized to false already.

from query.h

```
  Query()
      : d_description(""),	
        df_negate(false),	
        d_matchFunc(NULL),	
        d_dataFunc(NULL){};	
```

Reply:
So perhaps with the d_tol initialization in Query, we don't need the constructors anymore.